ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Stream.scala

```diff
@@ -73,18 +73,26 @@ private[h2] class H2Stream[F[_]: Concurrent](
 
   def sendMessageBody(mess: Message[F]): F[Unit] = {
     val trailers = mess.attributes.lookup(Message.Keys.TrailerHeaders[F])
-    mess.body.chunks.noneTerminate.zipWithNext
-      .foreach {
-        case (Some(c), Some(Some(_))) =>
-          sendData(c.toByteVector, false)
-        case (Some(c), Some(None) | None) =>
-          sendData(c.toByteVector, trailers.isEmpty)
-        case (None, _) =>
-          if (trailers.isDefined) Applicative[F].unit
-          else sendData(ByteVector.empty, true)
-      }
-      .compile
-      .drain
+    for {
+      maxFrameSize <- remoteSettings.map(_.maxFrameSize.frameSize)
+
+      _ <-
+        mess.body
+          .chunkLimit(maxFrameSize)
+          .noneTerminate
+          .zipWithNext
```

Comment:
On second look, I'm less sure about this fix. This code here seems to express a similar logic, but I'm not sure of the relationship between `writeWindow` and `maxFrameSize`.

https://github.com/http4s/http4s/blob/d7185b4867497b87bd0ee37c835fbb6cb3911868/ember-core/shared/src/main/scala/org/http4s/ember/core/h2/H2Stream.scala#L159-L169